src/lib/Gateway/ExpressionVisitor.php

```diff
@@ -226,63 +228,80 @@ private function expr(): ExpressionBuilder
         return $this->queryBuilder->expr();
     }
 
-    private function handleJoinQuery(
-        DoctrineSchemaMetadataInterface $relationshipMetadata,
-        string $field,
-        Comparison $comparison
+    private function handleRelationshipQuery(
+        DoctrineSchemaMetadataInterface $metadata,
+        DoctrineRelationshipInterface $relationship,
+        string $column,
+        Comparison $comparison,
+        ?QueryBuilder $subQuery
     ): string {
-        $tableName = $relationshipMetadata->getTableName();
-        $parameterName = $field . '_' . count($this->parameters);
-        $placeholder = ':' . $parameterName;
-
-        $value = $this->walkValue($comparison->getValue());
-        $type = $relationshipMetadata->getBindingTypeForColumn($field);
-        if (is_array($value)) {
-            $type += Connection::ARRAY_PARAM_OFFSET;
-        }
-
-        $parameter = new Parameter($parameterName, $value, $type);
+        $relationshipQuery = $this->relationshipTypeStrategyRegistry->handleRelationshipTypeQuery(
+            $metadata,
+            $relationship,
+            $column,
+            $comparison,
+            $subQuery ?? $this->queryBuilder,
+            $this->parameters
+        );
 
-        if (is_array($value)) {
-            $this->parameters[] = $parameter;
+        if ($relationship->getJoinType() === DoctrineRelationship::JOIN_TYPE_SUB_SELECT) {
+            if ($subQuery === null) {
+                throw new RuntimeMappingException(
```

Comment:
This is a logic exception. `ExpressionVisitor` should never reach this point without `$subQuery` argument, and there is nothing user can do about it, i.e. there would be no way to recover.
```suggestion
                throw new LogicException(
```

Additionally, this variable is not actually used - so this check seems... weird? It's more of a safety check that when operating on a subquery we actually received this argument, but at this point in the code we already acquired strategy from the registry?